components/brave_wallet/rust/lib.rs

```diff
@@ -215,30 +264,28 @@ fn bytes_are_curve25519_point(bytes: &[u8]) -> bool {
     curve25519_dalek::edwards::CompressedEdwardsY::from_slice(bytes).decompress().is_some()
 }
 
-fn decode_bech32(input: &str) -> Bech32DecodeResult {
-  if let Ok((hrp, data, variant)) = bech32::decode(&input) {
-    if let Ok(as_u8) = Vec::<u8>::from_base32(&data) {
-      match variant {
-        bech32::Variant::Bech32m => {
-          return Bech32DecodeResult {
-            hrp: hrp,
-            data: as_u8,
-            variant : Bech32DecodeVariant::Bech32m };
+fn decode_bech32(input: &str) -> Box<Bech32DecodeResult> {
+    let decoded = bech32::decode(&input);
+    match decoded {
+        Ok(decoded_value) => {
+            let (hrp, data, variant) = decoded_value;
+            Box::new(Bech32DecodeResult::from(
+                Vec::<u8>::from_base32(&data)
+                    .map_err(|err| Error::from(err))
```

Comment:
nit: This can just be `.map_err(Error::from)`. The closure takes the same arguments as the inner function, so it isn't doing anything.